File: 876.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    int getSize(ListNode* head){
        if (head->next == NULL) return 1; 
        int size = 0; 
        return getSize(head->next)+1; 
        
    }
    ListNode* middleNode(ListNode* head) {
        int middle = getSize(head)/2; 
        for (int i = 0; i < middle; i++){
            head = head->next; 
        }
        return head; 
    }
};

File: 876.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        linkedListSize = self.getLinkedListSize(head, 0) 
        
        return self.getNode(head, 0, linkedListSize//2) 
        
    def getLinkedListSize(self, head, size): 
        if head == None:
            return size 
        return self.getLinkedListSize(head.next, size+1)
    
    def getNode(self, head, pos, target_pos):
        if head == None: 
            return -1 
        if pos == target_pos:
            return head 
        return self.getNode(head.next, pos+1, target_pos)
